src/frequenz/client/base/channel.py

```diff
@@ -41,6 +42,24 @@ class SslOptions:
     """
 
 
+@dataclasses.dataclass(frozen=True)
+class KeepAliveOptions:
+    """Options for HTTP2 keep-alive pings."""
+
+    enabled: bool = True
+    """Whether keep-alive should be enabled."""
+
+    interval: timedelta | None = None
+    """The interval between pings.
+
+    If None, keep-alive is disabled."""
+
+    timeout: timedelta = timedelta(seconds=20)
+    """The time in milliseconds to wait for a keep-alive response.
+
+    This is only used if `interval` is not None and `enabled` is `True`."""
```

Comment:
This comment probably also needs update if you don't allow `None` for `interval`.

Reply:
done